File: Sortings/PythonSortings/QuickSort.py
def QuickSort(arr):
    """Функция быстрой сортировки"""
    if len(arr) <= 1:
        return
    base = arr[len(arr) // 2]
    L = []
    R = []
    M = []
    for i in range(len(arr)):
        if arr[i] < base:
            L.append(arr[i])
        elif arr[i] == base:
            M.append(arr[i])
        else:
            R.append(arr[i])

    QuickSort(L)
    QuickSort(R)
    k = 0
    for x in L+M+R:
        arr[k] = x
        k+=1
    

File: QuickSort/QuickSort/QuickSort.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace QuickSort
{
    public class QuickSort
    {
        public static void Sort(List<int> arr)
        {
            if(arr.Count <= 1)
                return;
            var left = new List<int>();
            var right = new List<int>();
            var eq = new List<int>();
            var barrier = arr[arr.Count / 2];
            for(int i = 0; i < arr.Count; i++)
            {
                if(arr[i] == barrier)
                    eq.Add(arr[i]);
                else if(arr[i] < barrier)
                    left.Add(arr[i]);
                else
                    right.Add(arr[i]);
            }
            Sort(left);
            Sort(right);
            left.AddRange(eq);
            left.AddRange(right);

            for(int i = 0; i < arr.Count; i++)
            {
                arr[i] = left[i];
            }

        }
    }
}
